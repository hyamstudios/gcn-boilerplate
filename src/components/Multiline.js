// @flow
import * as React from 'react'

type Props = {
  children: string,
  removeTrailingNewLines: boolean,
}

const Multiline = (props: Props) => {
  let arr: string[] = props.children.split('\n')
  if (props.removeTrailingNewLines) {
    const indexOfLastNonEmptyString: number = arr.reduce(
      (pointer, value, index) => {
        if (value !== '') {
          return index
        } else {
          return pointer
        }
      },
      0
    )
    arr = arr.slice(0, indexOfLastNonEmptyString + 1)
  }
  const a = arr.map((s, index, array) => (
    <React.Fragment key={index}>
      {s}
      {index === array.length - 1 ? false : <br />}
    </React.Fragment>
  ))
  return a
}

export default Multiline
