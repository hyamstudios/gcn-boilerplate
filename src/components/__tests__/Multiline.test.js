/* eslint-disable */

import React from 'react'
import Multiline from '../Multiline.js'
import { create } from 'react-test-renderer'

describe('component Multiline', () => {
  test('snapshot - no line break', () => {
    const component = create(<Multiline>{'1'}</Multiline>)
    expect(component.toJSON()).toMatchSnapshot()
  })
  test('snapshot - line breaks', () => {
    const component = create(<Multiline>{'1\n2\n3'}</Multiline>)
    expect(component.toJSON()).toMatchSnapshot()
  })
  test('snapshot - no trailing new lines', () => {
    const noTrailingNewLine = create(<Multiline>{'1\n2\n3\n'}</Multiline>)
    expect(noTrailingNewLine.toJSON()).toMatchSnapshot()
  })
})
