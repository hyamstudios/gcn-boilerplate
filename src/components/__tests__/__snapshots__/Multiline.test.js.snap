// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Multiline snapshot - line breaks 1`] = `
Array [
  "1",
  <br />,
  "2",
  <br />,
  "3",
]
`;

exports[`component Multiline snapshot - no line break 1`] = `"1"`;

exports[`component Multiline snapshot - no trailing new lines 1`] = `
Array [
  "1",
  <br />,
  "2",
  <br />,
  "3",
  <br />,
  "",
]
`;
